### PR TITLE
feat: add media provider

### PR DESCRIPTION
**Pull request by Irilith**
Add media provider using the Windows APIs to manage media output information, including metadata like title, artist, album title, and play status.

#### Output
- **`title`**: `string` - The title of the media.
- **`subTitle`**: `string` - The subtitle or additional information.
- **`trackNumber`**: `number` - The track number for albums or playlists (probably).
- **`artist`**: `string` - The name of the artist.
- **`albumTitle`**: `string` - The title of the album.
- **`isPlaying`**: `boolean` - Indicates whether the media is currently playing.
- **`isSpotify`**: `boolean` - Indicates whether the media is from Spotify.

Its only works in Windows as you may notice since this use Windows APIs

Preview:
https://github.com/user-attachments/assets/b85cf740-0ec6-4916-926a-d68ca45a170f

### CODE REVIEW

**Comment by Irilith** [comment]
its may close #73 , may not ,as it doesn't have pause/unpause, previous and next call

File: bun.lockb
Comment (Irilith) [inline bun.lockb:1]: i use bun, and accidentally commit this, sorry.

**Comment by lars-berger** [comment]
@HolbyFPV’s actively working on the same feature rn at [`feat/media-provider`](https://github.com/glzr-io/zebar/commits/feat/media-provider/). Will DM him and see if we can combine these 2 somehow. This one uses polling whereas the `feat/media-provider` implementation uses event listeners, so would favour towards the listeners approach since it updates immediately on changes

**Comment by HolbyFPV** [comment]
Thanks for this PR @Irilith 

As lars mentioned, I'm currently working through this feature.  I haven't gotten to the `client-api` side yet, so your code will come in handy.  If it's cool with you, we can close this PR and I'll credit you when I make my pull.

**Comment by Irilith** [comment]
Sure m8

**Pull request closed without merging**